tools/bumper/git.go

```diff
@@ -0,0 +1,175 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"os/exec"
+	"strings"
+
+	"github.com/google/go-github/v32/github"
+	"github.com/pkg/errors"
+	"golang.org/x/oauth2"
+)
+
+// gitHubManager does the Github operations handling.
+type gitHubManager struct {
+	// github client.
+	client *github.Client
+
+	// context needed for github api
+	ctx context.Context
+}
+
+func NewGitHubManager(token string) (*gitHubManager, error) {
+	ctx := context.Background()
+	ts := oauth2.StaticTokenSource(
+		&oauth2.Token{AccessToken: token},
+	)
+	tc := oauth2.NewClient(ctx, ts)
+
+	githubMgr := &gitHubManager{
+		client: github.NewClient(tc),
+		ctx:    ctx,
+	}
+
+	err := githubMgr.verifyClient()
+	if err != nil {
+		return nil, errors.Wrap(err, "Failed to verify connection with github client API")
+	}
+
+	return githubMgr, nil
+}
+
+func (mgr *gitHubManager) verifyClient() error {
+	//do a small sanity check
+	_, _, err := mgr.client.Users.Get(mgr.ctx, "")
+	if err != nil {
+		return errors.Wrapf(err, "Failed to get user from github client API")
+	}
+	return nil
+}
+
+func (mgr *gitHubManager) getCurrentReleaseTag(component component, repoDir string) (string, error) {
```

Comment:
All the functions in this module are quite complicated. Please add a proper doc string explaining what are they doing, what are the inputs, side effects and outputs

Reply:
I'll add description in the commit where I implement it then

Reply:
Done